enforcer-rules/src/main/java/com/google/cloud/tools/dependencies/enforcer/LinkageCheckerRule.java

```diff
@@ -117,11 +127,19 @@ public void execute(@Nonnull EnforcerRuleHelper helper) throws EnforcerRuleExcep
       }
 
       String projectType = project.getArtifact().getType();
-      if (dependencySection == DependencySection.DEPENDENCIES && !"jar".equals(projectType)) {
-        // When checking non-BOM project, not interested in non JAR artifact
-        return;
+      if (readingDependencyManagementSection) {
+        if (!"pom".equals(projectType)) {
+          logger.warn("A BOM should have packaging pom");
+          return;
+        }
+      } else {
+        if (UNSUPPORTED_NONBOM_PACKAGING.contains(projectType)) {
+          // Unless checking BOM project, not interested in pom artifact
```

Comment:
comment still says this is just about poms

Reply:
Removed comment. The variable name says enough.